app/partials/coinify-modal.pug

```diff
@@ -1,106 +1,60 @@
-    button.button-muted(ng-show="onStep('summary') && !error" ng-click="goTo('select-payment-medium')" translate="GO_BACK")
-    .flex-1.flex-end(ng-hide="status.loading")
-      button.button-muted.mrm(ng-click="$close() && goToOrderHistory()" translate="CLOSE" ng-show="onStep('trade-formatted') && bitcoinReceived")
-      button.button-muted.mrm(ng-click="close()" translate="CANCEL_ORDER" ng-hide="afterStep('summary') || onStep('summary') && error")
-      button.button-muted.mrm(ng-click="close() && goToOrderHistory()" translate="FINISH_LATER" ng-show="onStep('isx')")
-      div(ng-hide="afterStep('summary')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-country')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="nextStep()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('email')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="signup()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('accept-terms')")
-        button.button-primary(
-          ladda-translate="CONTINUE"
-          ng-click="confirmOrContinue()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled()"
-          ng-show="onStep('select-payment-medium')")
-        button.button-primary(
-          ladda-translate="CONFIRM"
-          ng-click="buy()"
-          data-style="expand-left"
-          ui-ladda="status.waiting"
-          ng-disabled="isDisabled() || fiatFormInvalid || status.gettingQuote"
-          ng-show="onStep('summary') && !error")
-      div(ng-show="onStep('trade-in-review') || onStep('summary') && error || onStep('trade-formatted') && !bitcoinReceived")
-        button.button-primary(translate="CLOSE" ng-click="cancel() && goToOrderHistory()")
-      div(ng-show="onStep('trade-formatted') && bitcoinReceived")
-        button.button-primary(translate="SEE_BITCOIN" ng-hide="inMobileBuy" ng-click="cancel()" ui-sref="wallet.common.transactions")
-        button.button-primary(translate="SEE_BITCOIN" ng-show="inMobileBuy" ng-click="exitToNativeTx()")
+  
+  div(ng-if="onStep('email')" ng-controller="CoinifyEmailController" ng-include="'partials/coinify/email.pug'")
+  div(ng-if="onStep('accept-terms')" ng-controller="CoinifySignupController" ng-include="'partials/coinify/signup.pug'")
+  div(ng-if="onStep('select-payment-medium')" ng-controller="CoinifyMediumController" ng-include="'partials/coinify/medium.pug'")
+  div(ng-if="onStep('summary')" ng-controller="CoinifySummaryController" ng-include="'partials/coinify/summary.pug'")
```

Comment:
If you move each step into its own component / directive, you could do something like this:
```
coinify-summary(on-complete='nextStep()', ng-if="onStep('summary')")
```
This CoinifyController would be responsible for figuring out the current step and which step is next. Each component should only care about check if it's step is complete and then call `scope.onComplete()`.

You can take it one step further with ui-router, which works inside modals:
https://github.com/angular-ui/ui-router/wiki/Frequently-Asked-Questions#how-to-open-a-dialogmodal-at-a-certain-state

Each step is then a route, e.g. `coinify.summary`, so you don't need `onStep()`. Instead, the CoinifyController is the parent state `coinify` and it takes care of sending the user to the next step (as well as sending the user back to a previous step if they try to nagivate to a future step too early).

Since both the individual step component and the CoinifyController need to know the criteria for when a step is completed, I think CoinifyController should have keep an array of function to track this. E.g.

```js
scope.isCompleted = [(a, b) => a & b]
```

Although at that point it might be better to use a service, and perhaps the routes file is a more appropriate place to handle this. @jtormey?